static/js/message_fetch.js

```diff
@@ -54,11 +54,7 @@ function process_result(data, opts) {
         message_util.add_old_messages(messages, opts.msg_list);
     }
 
-    if (
-        opts.msg_list === message_list.narrowed &&
-        message_lists.current === message_list.narrowed &&
-        message_list.narrowed.empty()
-    ) {
+    if (messages.length === 0 && message_lists.current.narrowed && message_lists.current.empty()) {
```

Comment:
This looks like an incorrect resolution of merge conflicts; we removed the `messages.length` check intentionally in 16c45aaf427a48af430eb6af8c9335d48c12c1ae.

I think we want to just do `opts.msg_list === message_lists.current && opts.msg_list.narrowed && msg.list.empty()`

Reply:
Ok, fixed!